### ISSUE TITLE
Add Jacobian support to JAX

### ISSUE THREAD

**Issue by agoose77** [scikit-hep/awkward#1772]
### Description of new feature

Although we aren't planning on further work here for a while, unless we find an easy path to do this, this issue tracks the problem of supporting JAX's Jacobian computation.

A few things to consider:

- JAX passes objects in addition to buffers to unflatten. We need to be able to pass these back to JAX, I believe. Doing this would probably require either a parameter flag, if we can return any-old-object, or a weak registry if not.
- The implementation of `jax.jacrev` explains how JAX actually calculates the Jacobian in pytree space. JAX understands the Jacobian as a 2D array, so if we re-interpret this 2D buffer for out layouts, we need to store sufficient metadata to be able to return it back to JAX for subsequent calculations, as it seems JAX goes to/from pytrees multiple times.
- See https://github.com/google/jax/discussions/12664 for a general discussion following this.

**Comment by agoose77** [comment]
@jakevdp we already had this issue open, so I'm going to use this as our tracker for the follow up of our meeting today!

**Comment by jpivarski** [comment]
@Saransh-cpp, please self-assign. (I'm adding the `autodiff` label.)
